pinot-segment-local/src/main/java/org/apache/pinot/segment/local/data/manager/TableDataManager.java

```diff
@@ -44,6 +44,12 @@
 @ThreadSafe
 public interface TableDataManager {
 
+  enum SegmentState {
```

Comment:
These are already defined.
Also, we don't need the definitions if we had individual methods for state transitions

Reply:
This has been reverted as described in related comment